webapp/src/components/ProgramActivityDialog.tsx

```diff
@@ -1,85 +1,212 @@
+  const [completed, setCompleted] = useState<boolean | null>(null);
+  const [isUpdateSuccess, setIsUpdateSuccess] = useState<boolean>(false);
+  const [isMessageVisible, setIsMessageVisible] = useState<boolean>(false);
+  const [error, setError] = useState<boolean | null>(null);
+  const [isErrorShown, setIsErrorShown] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  useEffect(() => {
+    async function fetchData() {
+      setCompleted(null);
+      if (show) {
+        await sendAPIGetRequest(
+          `/programs/activityStatus/${contents.programId}/${contents.curriculumActivityId}`,
+          contents.activityType,
+          setCompleted,
+          setError,
+          setIsErrorShown,
+          setIsLoading
+        );
+      }
+    }
+    fetchData();
+  }, [contents, show]);
+
+  const sendCompletedStatus = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    completed: boolean
+  ) => {
+    event.preventDefault();
+    if (event.type !== 'click') return;
+    sendAPIPutRequest(
+      `/programs/activityStatus/${contents.programId}/${contents.curriculumActivityId}`,
+      { completed: completed },
+      setCompleted,
+      setIsUpdateSuccess,
+      setIsMessageVisible,
+      setIsLoading
+    );
+  };
+
   const timeRange = () => {
     if (
       !(
         contents.start &&
         contents.end &&
         contents.description &&
-        contents.title
+        contents.title &&
+        contents.activityType
       )
     ) {
       return;
     }
     if (contents.allDay) {
       return formatDate(contents.start.toDateString());
     }
-
     let timeRangeString = '';
-
     timeRangeString += formatDate(contents.start.toString()) + ' ';
     timeRangeString += formatTime(contents.start.toString());
-
     if (contents.start.toString() === contents.end.toString()) {
       return timeRangeString;
     }
-
     timeRangeString += decodeHTML('&nbsp;&ndash;&nbsp;');
-
     if (contents.start.toDateString() !== contents.end.toDateString()) {
       timeRangeString += formatDate(contents.end.toString()) + ' ';
     }
     timeRangeString += formatTime(contents.end.toString());
-
     return timeRangeString;
   };
-
+  const tableHeaderCellStyle = {
+    fontWeight: 'bold',
+    border: 'none',
+    verticalAlign: 'top',
+    textAlign: 'right',
+  };
   return (
     <Dialog
       open={show}
       onClose={handleClose}
-      maxWidth="xs"
+      maxWidth="sm"
       aria-labelledby="alert-dialog-title"
       aria-describedby="alert-dialog-description"
     >
       <DialogTitle
         id="alert-dialog-title"
-        sx={{ textAlign: 'center', fontWeight: 'bold' }}
+        sx={{
+          textAlign: 'center',
+          fontWeight: 'bold',
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
       >
         {contents.title}
+        {completed && <TaskAltIcon sx={{ ml: 1 }} />}
```

Comment:
I can't reproduce Prabh's screenshot because that activity no longer exists in the database, but if you wanted a workaround for his issue that doesn't require adding a background to the IconButton (which you shouldn't because a. that's not a good solution, and b. makes it harder for us to support Dark Mode if we ever decide to do that in the future), you could have the child of `<DialogTitle>` be a `<Stack>` and handle it that way:

```typescript
      <DialogTitle id="alert-dialog-title" sx={{ fontWeight: 'bold' }}>
        <Stack
          direction="row"
          justifyContent="space-between"
          alignItems="center"
          spacing={2}
        >
          <div></div>
          <div>
            {contents.title}
            {completed && <TaskAltIcon sx={{ ml: 1 }} />}
          </div>
          <IconButton
            aria-label="close"
            onClick={handleClose}
            sx={{
              color: theme => theme.palette.grey[500],
            }}
          >
            <CloseIcon />
          </IconButton>
        </Stack>
      </DialogTitle>
```

But! Up to you to decide which is the best approach here:

1. The advantage to the `position: absolute` approach is that the close button is always in the top right corner no matter what, but the disadvantages are that it can overlap anything that's behind it, and putting a background behind it may hide the activity completion icon.
2. The advantage to the <Stack> approach is that the close icon won't appear on top of anything, but the disadvantage is that I have no idea how it will behave with a long title.
3. Do nothing, ship as is.